test/functional/cmdLineTests/jvmtitests/build.xml

```diff
@@ -51,11 +51,28 @@ SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0-only WITH Classpath-ex
 		<echo>===debug:                        on</echo>
 		<echo>===destdir:                      ${build}</echo>
 
+		<!-- Define variables to exclude Java21 and up tests for Java 8, 11 and 17. -->
+		<if>
+			<not>
+				<matches string="${JDK_VERSION}" pattern="^(8|11|17)$$" />
+			</not>
```

Comment:
Can you invert this `<if>` element so `<not>` is not needed here?

Reply:
It was initially inverted: see https://github.com/eclipse-openj9/openj9/pull/18856#discussion_r1527591235. @llxia suggested the `<not>` version since it was easier to read.

Reply:
I disagree, but I'm not sure it's worth changing it now.

I've been meaning for some time to replace most (all?) the regular expression matching of versions with uses of `<isgreaterthan>` or `<islessthan>` which are available in `ant-contrib`:
```
	<typedef name="isgreaterthan" classname="net.sf.antcontrib.logic.condition.IsGreaterThan" classpath="${antcontrib.jar}" />
	<typedef name="islessthan" classname="net.sf.antcontrib.logic.condition.IsLessThan" classpath="${antcontrib.jar}" />
```
The `<if>` element here would become (after inversion):
```
		<if>
			<islessthan arg1="${JDK_VERSION}", arg2="21" />
			<then>
				<property name="excludeJDK21UpGetStackTraceExtendedTest" value="com/ibm/jvmti/tests/getStackTraceExtended/gste002.java" />
				<property name="excludeJDK21UpGetThreadListStackTracesExtendedTest" value="com/ibm/jvmti/tests/getThreadListStackTracesExtended/gtlste002.java" />
			</then>
			<else>
				<property name="excludeJDK21UpGetStackTraceExtendedTest" value="" />
				<property name="excludeJDK21UpGetThreadListStackTracesExtendedTest" value="" />
			</else>
		</if>
```

See also https://github.com/antlibs/ant-contrib/blob/master/src/main/java/net/sf/antcontrib/logic/condition/IsLessThan.java.

Reply:
> It was initially inverted

Not the way I read that comment; the `<not>` simplified the expression of the condition, but it would have been further simplified by inverting the `<if>`.

Reply:
I misinterpreted your initial question. I get it know. Opened https://github.com/eclipse-openj9/openj9/pull/19188 to remove the `<not> ... </not>` block.